File: 백준/Silver/10828. 스택/스택.py
import sys

input = sys.stdin.readline

n = int(input())
stack = []

for _ in range(n):

    comment = input()
    if "push" in comment:
        push, num = comment.split()
        stack.append(int(num))

    elif "pop" in comment:
        if len(stack) == 0:
            print("-1")
        else:
            print(stack.pop())

    elif "size" in comment:
        print(len(stack))

    elif "empty" in comment:
        if len(stack) != 0:
            print(0)
        else:
            print(1)
    elif "top" in comment:
        if len(stack) != 0:
            print(stack[-1])
        else:
            print(-1)

File: 백준/Silver/10828. 스택/스택.java

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Stack;

public class Main {

	public static void main(String[] args) throws Exception {

		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		int n = Integer.parseInt(br.readLine());
		Stack<Integer> st = new Stack<Integer>();

		for (int i = 0; i < n; i++) {
			String[] split = br.readLine().split(" ");
			
			if (split[0].equals("top")) {
				if (st.empty()) {
					System.out.println(-1);
				} else {
					System.out.println(st.peek());
				}
			} else if (split[0].equals("size")) {
				System.out.println(st.size());
			} else if (split[0].equals("empty")) {
				if (st.isEmpty()) {
					System.out.println(1);
				} else {
					System.out.println(0);
				}
			} else if (split[0].equals("pop")) {
				if (st.isEmpty()) {
					System.out.println(-1);
				} else {
					System.out.println(st.pop());
				}
				
			} else {
				st.add(Integer.parseInt(split[1]));
			}
			
			
		}

	}

}
